api/files.py

```diff
@@ -165,3 +158,48 @@ def guess_type_from_filename(filename):
     else:
         filetype = None
     return filetype
+
+
+def get_valid_file(file_info):
+    file_id = file_info.get('_id', '')
+    file_hash = file_info.get('hash', '')
+    file_uuid_path = None
+    file_hash_path = None
+
+    if file_hash:
+        file_hash_path = util.path_from_hash(file_hash)
+
+    if file_id:
+        file_uuid_path = util.path_from_uuid(file_id)
+
+    if config.support_legacy_fs:
+        if file_hash_path and config.legacy_fs.isfile(file_hash_path):
+            return file_hash_path, config.legacy_fs
+        elif file_uuid_path and config.legacy_fs.isfile(file_uuid_path):
+            return file_uuid_path, config.legacy_fs
+
+    if file_uuid_path and config.fs.isfile(file_uuid_path):
+        return file_uuid_path, config.fs
+    else:
+        raise fs.errors.ResourceNotFound('File not found: %s', file_info['name'])
+
+
+def get_signed_url(file_path, file_system, filename=None):
+    try:
+        if type(file_system).__name__ in config.signed_url_available:
+            return file_system.geturl(file_path, filename=filename)
+    except fs.errors.NoURL:
+        return None
+
+
+def get_fs_by_file_path(file_path):
+    if config.support_legacy_fs:
+        if config.legacy_fs.isfile(file_path):
```

Comment:
Both if's test `config.legacy_fs.isfile(file_path)`...?

Reply:
thanks, I forgot to remove that